pkg/cloud/azure/services/virtualmachines/virtualmachines.go

```diff
@@ -246,6 +251,11 @@ func (s *Service) deriveVirtualMachineParameters(vmSpec *Spec, nic network.Inter
 		return nil, fmt.Errorf("failed to get Spot VM options %w", err)
 	}
 
+	dataDisks, err := generateDataDisks(vmSpec)
+	if err != nil {
+		return nil, err
```

Comment:
Please add context to this error
```suggestion
		return nil, fmt.Errorf("failed to generate data disk spec: %w", err)
```